Array 1.43.0 (#5025)
PostHog/posthog.com: contents/blog/the-posthog-array-1-43-0.md

---
date: 2023-01-16
title: "Array 1.43.0: Massive performance improvements!"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
---

Want to know more about what we're up to? Check out [our roadmap](/roadmap) to see what we're working on and what new beta features are available! You can also subscribe to [our Hogmail newsletter](/newsletter)!

## Update guide for self-hosted users

> Any notes about async migrations, etc. go here.

Author: @tiina303 Just tagging you here, as you've provided info on this for the last few Arrays

- [Improved: Recording playback controls](#improved-recording-playback-controls)
- [Improved: Browsing recordings and creating playlists](#improved-browsing-recordings-and-creating-playlists)

### New: Performance improvements

@Karl to add a sentence or two here. 

### New: Better insight searching
We don't just want searching for insights to be functional. We want it to be a world-class, joy-sparking event! So, we've updated search to include the description field and any tags which have been applied!

This sounds like a small change, but it makes a massive difference for teams that organize data well and generate a lot of insights!

### New: Feature flag variant overrides
Ever created a multivariate feature flag, and wanted to show the control variant to specific users? Or, maybe you found a new cohort you'd like to add to the test variant? Now you can!

1.43.0 adds the ability to  manually override variants via the flag edit screen. It's great for ensuring users get the experience you want, but is also useful for testing as it enables you to deterministically choose a variant, and test across client-side and server-side feature flags.

### New: Export recordings to file

> Sometimes a recording can be so insightful, so important that you think - I want to keep a record of this forever. Well now you can export any recording to a file and load it back into PostHog for playback in the future. Whether it is to commerate your first recorded signup or for compliance reasons, the tools are now in your hands.

Reviewer: ```suggestion
Sometimes a recording can be so insightful, so important that you think - I want to keep a record of this forever. Well now you can export any recording to a file and load it back into PostHog for playback in the future. Whether it is to commemorate your first recorded signup or for compliance reasons, the tools are now in your hands.
```

Until now! 1.43.0 adds calls into the dashboard and insight logic when an insight is abandoned, attempting to cancel the query. The benefit? Less resource hogging, faster queries for all!

### Improved: Recording playback controls 
The recording "seekbar" has been revamped to make it clearer in which sections the user was active, where tracked events occur and what time you are scrubbing to.

### Improved: Browsing recordings and creating playlists
Following up on our new Playlists feature, we focused on the UX of browsing and pinning recordings to be as quick and intuitive as possible. The recordings list now scrolls independently of the player and creating a playlists of pinned recordings is much faster.

### Other improvements & fixes

You think that's it? Not by a long shot! Version 1.43 also adds hundreds of other improvements and fixes, including...

- **Improvement:** [We've consolidated container build workflows and made changes to build and push multi-arch images](https://github.com/PostHog/posthog/pull/13543).

> - **Improvement:** We made [a simpler, faster process for querying](https://github.com/PostHog/posthog/pull/13529) `my_last_viewed_at`

Reviewer: ```suggestion
- **Improvement:** We made [a simpler, faster process for querying the recently viewed insights on your home page](https://github.com/PostHog/posthog/pull/13529)
```

## Share your feedback
We'd love to hear anything you have to say about PostHog, good or bad. As a thank you, we'll share some awesome [PostHog merch](https://merch.posthog.com).

Want to get involved? [Schedule a 30 minute call](https://calendly.com/posthog-feedback) with one of our teams to help us make PostHog even better!

## Contributions from the community
We always welcome contributions from our community and this time we want to thank the following people this month...

- @Ismaaa for [correcting some mistakes in our onboarding documentation](https://github.com/PostHog/posthog.com/pull/1840)
- @Johanholmerin for [fixing an issue when copying non-extensible objects](https://github.com/PostHog/posthog-js/pull/478)

Want to build with us? Check out our [contributing resources](/docs/contribute) to get started. We have a [list of Good First Issues](https://github.com/PostHog/posthog/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) for ideas on where you can contribute!

## Open roles at PostHog
We're currently hiring for remote candidates in the following role:

> - [Full Stack Engineer - Experimentation Team](/careers/full-stack-engineer-experimentation)

Reviewer: ```suggestion
- [Full Stack Engineer - Experimentation Team](/careers/full-stack-engineer-experimentation)
- [Sire Reliability Engineer](/careers/site-reliability-engineer)
```

## Share your feedback
We'd love to hear anything you have to say about PostHog, good or bad. As a thank you, we'll share some awesome [PostHog merch](https://merch.posthog.com).

Want to get involved? [Schedule a 30 minute call](https://calendly.com/posthog-feedback) with one of our teams to help us make PostHog even better!

## Contributions from the community
We always welcome contributions from our community and this time we want to thank the following people this month...

- @Ismaaa for [correcting some mistakes in our onboarding documentation](https://github.com/PostHog/posthog.com/pull/1840)
- @Johanholmerin for [fixing an issue when copying non-extensible objects](https://github.com/PostHog/posthog-js/pull/478)

Want to build with us? Check out our [contributing resources](/docs/contribute) to get started. We have a [list of Good First Issues](https://github.com/PostHog/posthog/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) for ideas on where you can contribute!

## Open roles at PostHog
We're currently hiring for remote candidates in the following role:

- [Full Stack Engineer - Experimentation Team](/careers/full-stack-engineer-experimentation)

> - [Sire Reliability Engineer](/careers/site-reliability-engineer)

Author: ```suggestion
- [Site Reliability Engineer](/careers/site-reliability-engineer)
```

Reviewer: 🤣 that makes more sense 😊

featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
---

Want to know more about what we're up to? Check out [our roadmap](/roadmap) to see what we're working on and what new beta features are available! You can also subscribe to [our Hogmail newsletter](/newsletter)!

## Update guide for self-hosted users

Any notes about async migrations, etc. go here.

## PostHog 1.43.0 release notes

**Release highlights:**

- [New: Performance improvements](#new-performance-improvements)
- [New: Role based access](#role-based-access)
- [New: Better insight searching](#new-better-insight-searching)
- [New: Feature flag variant overrides](#new-feature-flag-variant-overrides)
- [New: Export recordings to file](#new-export-recordings-to-file)
- [Improved: More experiment variants and improved flow](#improved-more-experiment-variants-and-improved-flow)
- [Improved: Query cancelling improvements](#improved-query-cancelling-improvements)
- [Improved: Recording playback controls](#improved-recording-playback-controls)
- [Improved: Browsing recordings and creating playlists](#improved-browsing-recordings-and-creating-playlists)

### New: Performance improvements

> @Karl to add a sentence or two here. 

Reviewer 2: ```suggestion
We don't want you to spend time waiting on machines so performance has had a renewed focus. Some improvements shipped this release:

- ClickHouse 22.8 support. This release also enables an important query optimization which will speed up some expensive queries up to 30%.
- Reworked dashboard cache refreshing logic - your dashboards should now be more up-to-date
- When updating dashboard filters we now may hit the cache for faster results
- Improved query cancellation when navigating away from trends or dashboards
- Reduced dashboard API response size. Existing `items` field is now deprecated and removed next release
- Property filters modal has received a speedup
- Home dashboards are no longer loaded twice
- When refreshing dashboards we now reload tiles in the same order as they're placed in visually
- When loading dashboards, we no longer load data from redis twice
- When updating dashboard filters, we now use cached data instead of always refreshing immediately
- Trends queries received an update which will reduce cross-shard traffic significantly and speed up queries
- Improved celery housekeeping tasks to require less ClickHouse resources. Run 0008 async migration to get full benefits of this.
- Many other small and large API speedups 

Last but not least, we now also have a [ClickHouse manual](/handbook/engineering/clickhouse) where we're gathering information we have learned about the database powering PostHog.
```

### New: Role based access

> We can now create roles and group team members together, along with being able to customize feature flags access for team members! Having access control helps reduce accidental changes and ensures confidence when shipping a new feature.

Reviewer 3: Happy to have some extra help in wording or phrasing here 😺

---
date: 2023-01-16
title: "Array 1.43.0: Massive performance improvements!"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
---

Want to know more about what we're up to? Check out [our roadmap](/roadmap) to see what we're working on and what new beta features are available! You can also subscribe to [our Hogmail newsletter](/newsletter)!

> **Need to update a self-hosted instance?** [Check the runbook docs for instructions](/docs/runbook/upgrading-posthog)!

## PostHog 1.43.0 release notes

**Release highlights:**

- [New: Performance improvements](#new-performance-improvements)
- [New: Feature flag variant overrides](#new-feature-flag-variant-overrides)
- [New: Export recordings to file](#new-export-recordings-to-file)
- [New: Better insight searching](#new-better-insight-searching)

> - [New: Show event/property counts in persons modal](#new-show-eventproperty-counts-in-persons-modal)

Reviewer 4: This doesn't point to a paragraph yet – just to be explicit, should I write one or would you prefer to take over this?

Author: Nope, I've written one and got a gif. Something is up with this PR which is giving me grief though. Let me try to add it in again.

### New: Role-based access for feature flags
You can now create roles and group team members together, along with customizing feature flags access for team members! Having access control helps reduce accidental changes and ensures confidence when shipping a new feature.

### Improved: Recording playback controls 
![recording controls](../images/blog/array/1-43-0-seekbar.gif)

We've revamped the recording seekbar to make it easier to find user activity at a glance, to show where tracked events occur and display what time you're scrubbing to.

### Other improvements & fixes
You think that's it? Not by a long shot! Version 1.43 also adds hundreds of other improvements and fixes, including...

- **Improvement:** [We've consolidated container build workflows and made changes to build and push multi-arch images](https://github.com/PostHog/posthog/pull/13543).
- **Improvement:** We made [a simpler, faster process for querying the recently viewed insights on your home page](https://github.com/PostHog/posthog/pull/13529)

> - **Improvement:** We've switched to use `pnpm` instead of `yarn` to [manage dependencies](https://github.com/PostHog/posthog/pull/13190).

Reviewer 4: I would put No. 1 and No. 3 at the bottom of the list, because they're honestly completely irrelevant to any PostHog user. It's just back office enhancements for PostHog engineering.

---
date: 2023-01-16
title: "Array 1.43.0: Massive performance improvements!"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true

> categories: ['Release notes', 'Product updates']

Reviewer 5: ```suggestion
category: PostHog news
tags:
  - Product updates
  - Release notes
```

---
date: 2023-01-16
title: "Array 1.43.0: Massive performance improvements!"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
category: PostHog news
tags:
  - Product updates
  - Release notes
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
---

Want to know more about what we're up to? Check out [our roadmap](/roadmap) to see what we're working on and what new beta features are available! You can also subscribe to [our Hogmail newsletter](/newsletter)!

> 

Author: ```suggestion
```

**Release highlights:**

> - [New: Performance improvements](#new-performance-improvements)

Reviewer 6: Also newwww incoming!: https://github.com/PostHog/posthog/pull/13611/files

Permissioning on feature flags, for enterprise

Reviewer 3: I've also created a manual docs for this, can someone take a look at it? :) https://github.com/PostHog/posthog.com/pull/5008
